src/workerd/io/compatibility-date.c++

```diff
@@ -296,4 +296,64 @@ kj::Maybe<kj::String> normalizeCompatDate(kj::StringPtr date) {
   return CompatDate::parse(date).map([](auto v) { return v.toString(); });
 }
 
+struct PythonSnapshotParsedField {
```

Comment:
For reviewers: this is a copy of the relevant code from downstream -- there will be a small refactor to make downstream use this version instead.